Sources/VLCRendererDiscovererManager.swift

```diff
@@ -0,0 +1,257 @@
+                        movieViewController.playingExternallyView.isHidden = true
+                    }
+                    playbackController.mediaPlayerSetRenderer(nil)
+                }
+                // Reset buttons state
+                for button in rendererButtons {
+                    button.isSelected = false
+                }
+            }
+            actionSheet.collectionView.reloadData()
+            actionSheet.updateViewConstraints()
+        }
+
+        // No more renderers to show
+        if getAllRenderers().isEmpty {
+            for button in rendererButtons {
+                button.isHidden = true
+            }
+        }
+    }
+}
+
+// MARK: VLCActionSheetDelegate
+extension VLCRendererDiscovererManager: VLCActionSheetDelegate {
+    func headerViewTitle() -> String? {
+        return NSLocalizedString("HEADER_TITLE_RENDERER", comment: "")
+    }
+
+    func itemAtIndexPath(_ indexPath: IndexPath) -> Any? {
+        let renderers = getAllRenderers()
+        if indexPath.row < renderers.count {
+            return renderers[indexPath.row]
+        }
+        return nil
+    }
+
+    func actionSheet(collectionView: UICollectionView, didSelectItem item: Any, At indexPath: IndexPath) {
+        guard let renderer = item as? VLCRendererItem,
+            let cell = collectionView.cellForItem(at: indexPath) as? VLCActionSheetCell else {
+                return
+        }
+        // Handles the case when the same renderer is selected
+        if renderer == VLCPlaybackController.sharedInstance().renderer {
+            cell.icon.image = UIImage(named: "rendererBlack")
+        } else {
+            // Reset all cells
+            collectionView.reloadData()
+            cell.icon.image = UIImage(named: "rendererBlackFull")
+            collectionView.layoutIfNeeded()
+        }
+    }
+}
+
+// MARK: VLCActionSheetDataSource
+extension VLCRendererDiscovererManager: VLCActionSheetDataSource {
+    func numberOfRows() -> Int {
+        return getAllRenderers().count
+    }
+
+    func actionSheet(collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(
+            withReuseIdentifier: VLCActionSheetCell.identifier, for: indexPath) as? VLCActionSheetCell else {
+                return UICollectionViewCell()
+        }
+        let renderers = getAllRenderers()
+        if indexPath.row < renderers.count {
+            let rendererName = renderers[indexPath.row].name
+            cell.name.text = rendererName
+
+            cell.icon.image = UIImage(named: "rendererGray")
+            if renderers[indexPath.row] == VLCPlaybackController.sharedInstance().renderer {
+                cell.icon.image = UIImage(named: "rendererFullOrange")
+            }
+        } else {
+            assertionFailure("VLCRendererDiscovererManager: cellForItemAt: IndexPath out of range")
```

Comment:
👍 but we still fall through here and return the cell in release is this okay ?

Reply:
Would a `fatalError` be better?